beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -245,3 +254,23 @@ func (s *Store) tips() ([][32]byte, []types.Slot) {
 	}
 	return roots, slots
 }
+
+// HighestReceivedBlockSlot returns the highest slot received by the forkchoice
+func (f *ForkChoice) HighestReceivedBlockSlot() types.Slot {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	return f.store.highestReceivedSlot
+}
+
+// ReceivedBlocksLastEpoch returns the number of blocks received in the last epoch
+func (f *ForkChoice) ReceivedBlocksLastEpoch(slot types.Slot) uint64 {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	count := uint64(0)
+	for _, s := range f.store.receivedBlocksLastEpoch {
+		if s != 0 && slot-params.BeaconConfig().SlotsPerEpoch < s {
```

Comment:
this can underflow, better pass `SlotsPerEpoch` to the other side of the inequality

Reply:
Also, same as above: why not use the wall clock?